### DISCUSSION TITLE
How to make Android Remote Control find devices

### DISCUSSION THREAD

**Discussion by elliotn** [rern/rAudio#746] [Tips]
FYI -

The RuneAudio Android Remote Control uses service discovery to find RuneAudio devices. Specifically, it expects
the service to respond with the following regex that includes the hostname and the mac address:

  private static final String SERVICE_NAME = "^.* * \\[(.*)\\] \\[(.*)\\]$";

Update /etc/avahi/services/ros.service with proper values, and the remote will find the device. E.g.:
  <name replace-wildcards="yes">RuneAudio [host] [xx:xx:xx:xx:xx:xx]</name>

**Comment by pepper7777** [comment]
hi elliotn
Can you please share complete content of the file as I'm not sure how to modify it.
thanks
Pepper

**Reply by elliotn** [reply]
Sure:

```
<?xml version="1.0" standalone='no'?>
<!DOCTYPE service-group SYSTEM "avahi-service.dtd">
<service-group>
<name replace-wildcards="yes">RuneAudio [%h] [xx:xx:xx:xx:xx:xx]</name>
  <service>
    <type>_http._tcp</type>
    <port>80</port>
  </service>
</service-group>

```
----

Essentially the `<name replace-wildcards>`   line is made up of 3 items:
 - RuneAudio
 - host name (above %h is replaced with the actual name. this could be any text, e.g. living room)
 - mac address (I just put xx:xx:xx:xx:xx:xx above, as I don't really care about the actual mac address)

This is what it ends up looking like in the player:

![p](https://user-images.githubusercontent.com/680037/149672700-70a43c80-8a1b-429f-838d-3c47cd24e974.png)

**Reply by pepper7777** [reply]
Awsome.
Thank you :-)
